java/api/src/main/java/com/exadel/frs/core/trainservice/service/FaceVerificationProcessServiceImpl.java

```diff
@@ -47,7 +45,16 @@ public VerifyFacesResponse processImage(ProcessImageParams processImageParams) {
 
         VerifyFacesResponse result = getResult(findFacesResults.get(0), findFacesResults.get(1));
 
-        validateResult(result, processImageParams.getFacePlugins());
+        VerifyFacesResponse response = result.prepareResponse(result, processImageParams.getFacePlugins());
+        return cleanupResult(response, !processImageParams.getStatus());
+    }
+
+    private VerifyFacesResponse cleanupResult(VerifyFacesResponse result, boolean shouldClean) {
```

Comment:
Why don't you add it to `prepareResponse` method? Just pass another parameter - `boolean shouldClean` and do it there

Reply:
cleanupResult and prepareResult combined